### ISSUE TITLE
Make it available as a Docker container

### ISSUE THREAD

**Issue by vemonet** [dice-group/LSQ#26]
Hi, I am not sure if there are still development on this project. But if you want to make it easier for people to use your weird Jar file that does not run on windows: publish a docker image. 

It is **really easy** to include a jar file in a docker image and publish it for free (it's 2021, not 2009 anymore)

It will solve all your issues about distributing your software: will work out of the box on macos, windows, linux, can be easily deployed on clusters, etc

And it will avoid everyone who mostly just want to execute LSQ to need to git clone + mvn build (which takes a lot of time, is error prone)

You will not need to write and maintain multiple weird documentations and process to bundle with maven, then build a deb file... http://lsq.aksw.org/v2/setup.html

So you will take maybe 1 day to produce the docker image which will save you multiple days in the future

One cross platform solution: docker (anyway LSQ is not something used everyday so you dont need to make a deb package out of it, calling it with a docker run fits well it's use-cases)

**Comment by Aklakan** [comment]
I totally agree that there should be a docker container. One reason this hasn't happened yet that for the past months i wanted to transition the rdfize/analyze part of the processing to apache spark - but I only got around to do it one or two weeks ago.
Also the reason why the 'old' lsq is not cross-platform is because if its dependency on /usr/bin/sort - with the transition to spark lsq should be fully cross platform - in any case, a docker container certainly makes things alot easier.

**Comment by vemonet** [comment]
In a linux container you'll have access to `/usr/bin/sort` anywhere :) 

I created a pull request with a Dockerfile, and a GitHub Actions workflow to automatically build and publish the image to AKSW container registry on GitHub: https://github.com/AKSW/LSQ/pull/29


For Spark I am not sure how you plan to use it. I guess by connecting to an existing Spark cluster, but I added some commented lines to setup Spark in the docker image, in case you want to use it directly as standalone within the image

**Comment by Aklakan** [comment]
Revisiting this issue, there is for some time now the docker image which is built and [deployed](https://hub.docker.com/r/aksw/lsq) with the jib-maven-plugin. Obtainable with `docker pull aksw/lsq`. Files on the host can be bind mounted and provided as arguments.
